pkg/xds/experimental/client/cell.go

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+package client
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+)
+
+var Cell = cell.Module(
+	"xds-client",
+	"xDS client",
+
+	//exhaustruct:ignore
+	cell.Config(Config{Options: *Defaults}),
+	cell.Provide(NewXDSClient),
+)
+
+type Config struct {
+	Addr string
+
+	ProjectNum int64
+	Mesh, Zone string
+
+	Options
+}
+
+func (conf Config) Flags(flags *pflag.FlagSet) {
+	flags.String("xds-server-address", conf.Addr, "Address of xDS server")
+	flags.Bool("xds-use-sotw-protocol", conf.UseSOTW, "Use State Of The World, non-incremental version of xDS protocol")
+	flags.Int64("xds-node-project-num", conf.ProjectNum, "ProjectID of GCP project owning the mesh to join")
```

Comment:
This seems pretty specific to GCP, is there anything we can do to make it more generic?

Maybe it's better to just have the Cilium Cell take a NodeID string instead? It seems like all the ProjectNum, Mesh, and Zone flags are doing is being concatenated anyway?

Reply:
+1 Alexander -- we can't have any cloud provider specific things in here.

What is the generic XDS way to specify the URL to get resources from?